src/pymodaq/utils/parameter/pymodaq_ptypes/itemselect.py

```diff
@@ -71,28 +82,35 @@ def set_value(self, values):
             *values*          dictionnary    the values dictionnary to be setted.
             =============== ============== =======================================
         """
-        allitems = [item.text() for item in self.all_items()]
-        if allitems != values['all_items']:
-            self.clear()
-            for value in values['all_items']:
-                item = QtWidgets.QListWidgetItem(value)
-                if self.hasCheckbox:
+        allitems = []
+        # Check existing items
+        for item in self.all_items():     
+            if item.text() not in values['all_items']: # Remove items from list if text not in values
+                item = self.takeItem(self.row(item))
+                del item # Qt recommand to delete the item when removed
```

Comment:
we are in the python world so the unused things are garbage collected (no use to delete them manually)